Sources/Conduit/Auth/OAuth2TokenGrantManager.swift

```diff
@@ -55,7 +55,12 @@ struct OAuth2TokenGrantManager {
         return newToken
     }
 
-    static func errorFrom(data: Data?, response: HTTPURLResponse?) -> Error? {
+    static func errorFrom(data: Data?, response: HTTPURLResponse?, error: Error? = nil) -> Error? {
+        if let error = error as? NSError,
```

Comment:
Instead of checking error.code < 0, we check if the error belongs to NSURLErrorDomain, which groups all network-related errors.

```suggestion
     if let error = error as? NSError, error.domain == NSURLErrorDomain {
    return OAuth2Error.networkFailure
}
```